File: CSES/C++/Introductory Problems/Coin Piles.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>

typedef long long ll;



void f(ll x, ll y)
{
    if((x+y)%3==0)
    {
        ll sum = (x + y) / 3; 
        ll n = x - sum; 
        ll k = y - sum; 
        if ( x == 2*n + k && y == 2 * k + n && k>=0 && n>=0)
            std::cout << "YES" << std::endl;
        else
            std::cout << "NO" << std::endl;
    }
    else
        std::cout << "NO" << std::endl;
}


int main()
{
    ll in,x,y;
    std::cin >> in;
    for (ll i = 0; i < in; i++)
    {
        std::cin >> x >> y;
        f(x, y);
    }
}

File: CSES/Python/Introductory Problems/Coin Piles.py
def f(x:int, y:int):
    a = x+y
    if a % 3 != 0:
        print("NO")
    else:
        sum = int((x+y) // 3) 
        n = int(y - sum) 
        k = int(x - sum)
        if 2 * n + k == y and 2 * k + n == x and k >= 0 and n >=0 :
            print("YES")
        else: 
            print("NO")


n = int(input())
inputs = []
for i in range(n):
    inputs.append(input())

for i in range(n):
    x, y = map(int,inputs[i].split())
    f(x,y)